src/main/java/tasks/Task.java

```diff
@@ -0,0 +1,94 @@
+package tasks;
+
+import parser.Parser;
+import exceptions.ArgumentException;
+
+import java.time.LocalDate;
+
+public abstract class Task {
+    protected String description;
+    protected boolean isDone;
+
+    public static Task makeTask(String[] taskData) throws ArgumentException {
+        String taskType = taskData[0];
+        switch (taskType) {
+        case "T":
```

Comment:
I like the use of the `switch` statement here! As well as no indentation for `case` clauses. 👍